flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/sqlexec/SqlToOperationConverterTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sqlexec;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.EnvironmentSettings;
+import org.apache.flink.table.api.SqlDialect;
+import org.apache.flink.table.api.internal.TableEnvironmentImpl;
+import org.apache.flink.table.catalog.CatalogTable;
+import org.apache.flink.table.delegation.Planner;
+import org.apache.flink.table.operations.CatalogSinkModifyOperation;
+import org.apache.flink.table.operations.Operation;
+import org.apache.flink.table.operations.ddl.CreateTableOperation;
+import org.apache.flink.table.planner.operations.SqlConversionException;
+import org.apache.flink.table.types.DataType;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/** Test cases for SqlExecutableStatement. **/
+public class SqlToOperationConverterTest {
+	private static final EnvironmentSettings settings = EnvironmentSettings.newInstance()
```

Comment:
We should reduce the dependencies of tests to a minimum. Why does this test needs access to a `TableEnvironment`? Could you think of a test that really just tests the `SqlToOperationConverter`?

Reply:
Because the `SqlToOperationConverter ` needs the `FlinkPlannerImpl` which usually comes from a TableEnvironment, also i want to test that the `TableConfig#setSqlDialect` did work well. There is no much overhead here.

Reply:
There is the overhead of discovering a planner and executor. The test could also be based on the `Planner` only with a mocking catalog. We should always reduce the dependencies of tests to a minimum because otherwise we need to change many tests if we perform changes to the API class. This test must not use the API but can go at least one level deeper.

Reply:
Agreed and updated, thanks.